src/sentry/migrations/0289_dashboardwidgetquery_convert_orderby_to_field.py

```diff
@@ -0,0 +1,136 @@
+
+    if not match[2]:
+        return match[1]
+
+    arguments = parse_arguments(match[1], match[2])
+    alias = match[1] + "_" + "_".join(arguments)
+
+    for pattern in [r"[^\w]", r"^_+", r"_+$"]:
+        alias = re.sub(pattern, "_", alias)
+
+    return alias
+
+
+def parse_arguments(function: str, columns: str):
+    """
+    Some functions take a quoted string for their arguments that may contain commas,
+    which requires special handling.
+    This function attempts to be identical with the similarly named parse_arguments
+    found in static/app/utils/discover/fields.tsx
+    """
+    if (function != "to_other" and function != "count_if" and function != "spans_histogram") or len(
+        columns
+    ) == 0:
+        return [c.strip() for c in columns.split(",") if len(c.strip()) > 0]
+
+    args = []
+
+    quoted = False
+    escaped = False
+
+    i, j = 0, 0
+
+    while j < len(columns):
+        if i == j and columns[j] == '"':
+            # when we see a quote at the beginning of
+            # an argument, then this is a quoted string
+            quoted = True
+        elif i == j and columns[j] == " ":
+            # argument has leading spaces, skip over them
+            i += 1
+        elif quoted and not escaped and columns[j] == "\\":
+            # when we see a slash inside a quoted string,
+            # the next character is an escape character
+            escaped = True
+        elif quoted and not escaped and columns[j] == '"':
+            # when we see a non-escaped quote while inside
+            # of a quoted string, we should end it
+            quoted = False
+        elif quoted and escaped:
+            # when we are inside a quoted string and have
+            # begun an escape character, we should end it
+            escaped = False
+        elif quoted and columns[j] == ",":
+            # when we are inside a quoted string and see
+            # a comma, it should not be considered an
+            # argument separator
+            pass
+        elif columns[j] == ",":
+            # when we see a comma outside of a quoted string
+            # it is an argument separator
+            args.append(columns[i:j].strip())
+            i = j + 1
+        j += 1
+
+    if i != j:
+        # add in the last argument if any
+        args.append(columns[i:].strip())
+
+    return [arg for arg in args if arg]
+
+
+def convert_dashboard_widget_query_orderby_to_function_format(apps, schema_editor):
+    DashboardWidgetQuery = apps.get_model("sentry", "DashboardWidgetQuery")
+
+    for query in RangeQuerySetWrapperWithProgressBar(DashboardWidgetQuery.objects.all()):
+        if query.orderby == "":
+            continue
+
+        orderby = query.orderby
+        stripped_orderby = orderby.lstrip("-")
+        orderby_prefix = "-" if orderby.startswith("-") else ""
+
+        for aggregate in query.aggregates:
+            alias = get_function_alias(aggregate)
+            if alias == stripped_orderby:
```

Comment:
To confirm, I know queries support >1 orderby, but I see `DashboardWidgetQuery` has orderby as a string, this is cause `DashboardWidgetQuery` only supports exactly 1 orderby right?